### DISCUSSION TITLE
How to avoid firing an element mouseover, when I hover my mouseover an element tool?

### DISCUSSION THREAD

**Discussion by IshaanShettigar** [clientIO/joint#2358] [Q&A] [answered]
### Introduction

I am trying to achieve some custom behaviour whenever the user hovers over an element. So I am listening to the `element:mouseover` and `element:mouseout` events where I start and stop some behaviour. The problem I am facing is that when I hover over the elements tools, it fires the `element:mouseover` event which is not what I want. I have 4 resize tools and a rotate tool on each element. Do we need to call some `e.stopPropogation()` but I'm not sure where as I don't think element tool on hover is an event that is fired. Please help me out

Below is the sample code for the RotateTool
```
const setPositionAll = (view, coordinates, getPos) => {
    const { model } = view;
    const { width, height } = model.size();
    // Find the center of the element
    const center = new g.Point(width / 2, height / 2);
    // Find the angle between the current mouse coordinates and position of the rotate control tool, while taking the centre as a reference
    const angle = center.angleBetween(coordinates, getPos(view));
    model.rotate(Math.round(angle));
}


/**
 * @param {joint.dia.ElementView} view Element View
 * @returns centre of the model
 */
const getPositionIWST = (view) => {
    const { model } = view;
    const { width, height } = model.size()
    return new g.Point(width / 2, -height / 1.2);
}

/**
 * It uses the setPositionAll function and the getPositionIWST function
 */
const RotateToolIWST = joint.elementTools.Control.extend({
    children: rotateChildren,
    getPosition: function (view) {
        return getPositionIWST(view)
    },
    setPosition: function (view, coordinates) {
        setPositionAll(view, coordinates, this.getPosition)
    }
});
```

Here is code for my ResizeTool
```
const ResizeToolBottomRightST = joint.elementTools.Control.extend({
    getPosition: function (view) {
        // console.log("view", view);
        const model = view.model;
        const { width, height } = model.size()
        return { x: width + 0.25 * width + 10, y: height + 0.25 * height + 10 };

    },
    setPosition: function (view, coordinates) {
        const model = view.model;
        const { width, height } = model.size()
        model.resize(
            Math.max(coordinates.x - 0.25 * width - 10, 1),
            Math.max(coordinates.y - 0.25 * height - 10, 1)
        );
    },
});
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Have you tried `element:mouseenter` and `element:mouseleave` events?

See the difference between `mousenter` and `mouseover` events [here](https://www.w3schools.com/jquery/tryit.asp?filename=tryjquery_event_mouseenter_mouseover#:~:text=mouseenter%20and%20mouseover.-,The%20mouseover%20event%20triggers%20when%20the%20mouse%20pointer%20enters%20the,pointer%20enters%20the%20div%20element).

**Reply by IshaanShettigar** [reply]
I tried that but it didn't work as I wanted it to, but I see your point. The problem is, this is how my element is 
![image](https://github.com/clientIO/joint/assets/77607172/dcb518bb-46df-42b5-9046-0e5c789275ae)
When the cursor leaves element and then goes to the tool to resize/rotate it fires the `mouseenter` event again as it is entering the element.

**Reply by kumilingus** [reply]
You mean you don't want to show the resize buttons when you hover over the rotate button?
It would be best to create a JSFiddle with this problem (so we can better understand the problem and provide a solution faster).

**Reply by IshaanShettigar** [reply]
Basically what i wanted was, when the user hovers over the element it displays a small modal like this.
![image](https://github.com/clientIO/joint/assets/77607172/8128e423-eb32-4ac6-b61e-ce4de6e34446)
or this
![image](https://github.com/clientIO/joint/assets/77607172/fd233138-9952-4e94-afaf-73d1953cb933)

But the problem was when the user tried to resize/rotate they would click either of the tools. This triggered the `element:mouseenter` event. Which was not needed. I managed to solve my problem through another method (I just disabled the modal when the tools were shown), but my question still stands. How to avoid an element tool triggering an `element:mouseenter` event.

**Reply by kumilingus** [reply, marked as answer]
Triggering events by paper cannot be prevented. Similarly, you cannot force the browser to stop triggering mouse events.

What you can do, however, is ignore the event based on a certain condition.
